htdocs/core/class/html.form.class.php

```diff
@@ -1188,12 +1188,17 @@ function select_remises($selected, $htmlname, $filter, $socid, $maxvalue=0)
         // On recherche les remises
         $sql = "SELECT re.rowid, re.amount_ht, re.amount_tva, re.amount_ttc,";
         $sql.= " re.description, re.fk_facture_source";
+        if (!empty($conf->global->MAIN_SHOW_FACNUMBER_IN_DISCOUNT_LIST)) $sql.= ", f.facnumber";
         $sql.= " FROM ".MAIN_DB_PREFIX ."societe_remise_except as re";
+        if (!empty($conf->global->MAIN_SHOW_FACNUMBER_IN_DISCOUNT_LIST)) $sql.= " LEFT JOIN ".MAIN_DB_PREFIX."facture f ON (f.rowid = re.fk_facture_source)";
         $sql.= " WHERE re.fk_soc = ".(int) $socid;
         $sql.= " AND re.entity = " . $conf->entity;
         if ($filter) $sql.= " AND ".$filter;
         $sql.= " ORDER BY re.description ASC";
 
+        // Prevent sql error on the ambiguous column
```

Comment:
I have a better solution to suggest:
Because the select return always a small list, i suggest to include in previous select only the field re.fk_facture_source (always)
So you don't need to have a left join and no need to clean request

Then into loop, just do, if option is on, a fetch on invoice from the rowid to get the ref.